File: android-exercises/5 - list_view/1/ListView_java/app/src/main/java/com/bugrahankaramollaoglu/listview_java/MainActivity.java
package com.bugrahankaramollaoglu.listview_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Toast;

import com.bugrahankaramollaoglu.listview_java.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        // 1. create list for items
        String[] colors = {"pink","red","green","blue"};

        // 2. adapter
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, colors);

        // 3. link adapter with widget
        binding.listView.setAdapter(adapter);

        // 4. create onClick for items
        binding.listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                String selectedItem = adapter.getItem(position);
                Toast.makeText(MainActivity.this, "you selected: " + selectedItem, Toast.LENGTH_SHORT).show();
            }
        });

        // 5. that's it
    }
}

File: android-exercises/5 - list_view/1/ListView_kotlin/app/src/main/java/com/bugrahankaramollaoglu/listview_kotlin/MainActivity.kt
package com.bugrahankaramollaoglu.listview_kotlin

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bugrahankaramollaoglu.listview_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var myAdapter: ArrayAdapter<String>
    private var colors = listOf<String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        colors = listOf("red", "green", "blue", "purple", "pink")

        myAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, colors)

        binding.listView.adapter = myAdapter

        binding.listView.setOnItemClickListener { parent, view, position, id ->
            val selectedItem = myAdapter.getItem(position)
            Toast.makeText(this, "you chose $selectedItem", Toast.LENGTH_SHORT).show()
        }
    }
}